lib/Int64.fram

```diff
@@ -0,0 +1,71 @@
+pub method equal {self : Int64} = (extern dbl_eqInt64  : Int64 -> Int64 -> Bool) self
+pub method neq   {self : Int64} = (extern dbl_neqInt64 : Int64 -> Int64 -> Bool) self
+pub method gt    {self : Int64} = (extern dbl_gtInt64  : Int64 -> Int64 -> Bool) self
+pub method lt    {self : Int64} = (extern dbl_ltInt64  : Int64 -> Int64 -> Bool) self
+pub method ge    {self : Int64} = (extern dbl_geInt64  : Int64 -> Int64 -> Bool) self
+pub method le    {self : Int64} = (extern dbl_leInt64  : Int64 -> Int64 -> Bool) self
+
+pub method toString = (extern dbl_int64ToString : Int64 -> String) self
+
+pub method neg {self : Int64} = (extern dbl_negInt64 : Int64 -> Int64) self
+pub method add {self : Int64} = (extern dbl_addInt64 : Int64 -> Int64 -> Int64) self
+pub method sub {self : Int64} = (extern dbl_subInt64 : Int64 -> Int64 -> Int64) self
+pub method mul {self : Int64} = (extern dbl_mulInt64 : Int64 -> Int64 -> Int64) self
+
+pub let zero = 0L
+pub let one  = 1L
+pub let minus_one = (1L).neg
+pub method succ {self : Int64} = self + one
+pub method pred {self : Int64} = self - one
+
+pub method div {`re : {type X} -> Unit ->[|_] X} (n : Int64) =
+  if n.equal zero then `re ()
+  else (extern dbl_divInt64 : Int64 -> Int64 -> Int64) self n
+
+pub method mod {`re : {type X} -> Unit ->[|_] X} (n : Int64) =
+  if n.equal zero then `re ()
+  else (extern dbl_modInt64 : Int64 -> Int64 -> Int64) self n
+
+pub method land = (extern dbl_andInt64 : Int64 -> Int64 -> Int64) self
+pub method lor  = (extern dbl_orInt64  : Int64 -> Int64 -> Int64) self
+pub method lxor = (extern dbl_xorInt64 : Int64 -> Int64 -> Int64) self
+
+pub method shiftl  = (extern dbl_lslInt64 : Int64 -> Int64 -> Int64) self
+pub method shiftr  = (extern dbl_lsrInt64 : Int64 -> Int64 -> Int64) self
+pub method ashiftr = (extern dbl_asrInt64 : Int64 -> Int64 -> Int64) self
+
+pub let int64_of_int = (extern dbl_int64ToInt : Int -> Int64)
+
+
+
+
+pub method to_string {`re : {type X} -> Unit ->[|_] X, self : Int64} =
+	if self < 0x7fffffffffffffffL then
+		(extern dbl_int64ToString : Int64 -> String) self
+	else `re ()
+
+pub method to_string_opt {self : Int64} =
+	if self < 0x7fffffffffffffffL then
+		Some ((extern dbl_int64ToString : Int64 -> String) self)
+	else None
+
+pub method to_int {`re : {type X} -> Unit ->[|_] X, self : Int64} =
+	if self < 0x7fffffffffffffffL then
+		(extern dbl_intToInt64 : Int64 -> Int) self
+	else `re ()
+
+pub method to_int_opt {self : Int64} =
+	if self < 0x7fffffffffffffffL then
+		Some ((extern dbl_intToInt64 : Int64 -> Int) self)
+	else None
+
+pub method abs {self : Int64} =
+	if self < 0L then self.neg else self
```

Comment:
Why `self.neg` instead of `- self`?

Reply:
because it returns error:
```
> method neg {self : Int} = - self ;;
<stdin>:1:27: fatal error: Unbound variable (- .)
```
i dont know if we have unary minus, and with how operators are implemented if we can have it along side binary one

Reply:
Unary minus is supported, but I don't see it in the Prelude. It can be implemented in the following way:
```
method neg {self : Int} = 0 - self
method fn (- .) = neg
```